docker/Dockerfile

```diff
@@ -161,68 +162,48 @@ RUN --mount=type=bind,source=tower-batch-control,target=tower-batch-control \
     --mount=type=cache,target=${APP_HOME}/target/ \
     --mount=type=cache,target=/usr/local/cargo/git/db \
     --mount=type=cache,target=/usr/local/cargo/registry/ \
-cargo build --locked --release --features "${FEATURES}" --package zebrad --bin zebrad && \
-cp ${APP_HOME}/target/release/zebrad /usr/local/bin
+    cargo build --locked --release --features "${FEATURES}" --package zebrad --bin zebrad && \
+    cp ${APP_HOME}/target/release/zebrad /usr/local/bin
 
-# This stage is only used when deploying nodes or when only the resulting zebrad binary is needed
-#
-# To save space, this step starts from scratch using debian, and only adds the resulting
-# binary from the `release` stage
+# This step starts from scratch using Debian and only adds the resulting binary
+# from the `release` stage.
 FROM debian:bookworm-slim AS runtime
 
-# Set the default path for the zebrad binary
-ARG APP_HOME
-ENV APP_HOME=${APP_HOME}
-WORKDIR ${APP_HOME}
-
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
-    ca-certificates \
-    curl \
-    rocksdb-tools \
-    gosu \
-    && rm -rf /var/lib/apt/lists/* /tmp/*
+COPY --from=release /usr/local/bin/zebrad /usr/local/bin/
+COPY --from=release /usr/local/bin/entrypoint.sh /usr/local/bin/
 
-# Create a non-privileged user that the app will run under.
-# Running as root inside the container is running as root in the Docker host
-# If an attacker manages to break out of the container, they will have root access to the host
-# See https://docs.docker.com/go/dockerfile-user-best-practices/
-ARG USER=zebra
+# Create a non-privileged user for running `zebrad`.
+#
+# ## Security
+#
+# If an attacker manages to break out of the container, they will have root
+# access to the host. [1][1]
+#
+# [1]: https://docs.docker.com/go/dockerfile-user-best-practices/
+ARG USER="zebra"
 ENV USER=${USER}
-ARG UID=10001
-ENV UID=${UID}
-ARG GID=10001
-ENV GID=${GID}
-
-RUN addgroup --system --gid ${GID} ${USER} \
-    && adduser \
-    --system \
-    --disabled-login \
-    --shell /bin/bash \
-    --home ${APP_HOME} \
-    --uid "${UID}" \
-    --gid "${GID}" \
-    ${USER}
-
-# Config settings for zebrad
-ARG FEATURES
-ENV FEATURES=${FEATURES}
```

Comment:
If the `runtime` stage is built with custom `FEATURES` this will be propagated by default as an `ENV` variable to the `entrypoint.sh`, if we remove this instructions, then we need to always specify the `FEATURES` environment variable when running the image, otherwise it will be empty.

Reply:
Yes, but that var is not used in any meaningful way in `entrypoint.sh`, so I removed it. I would like to refactor the way we configure Zebra inside Docker in follow-up PRs.

Reply:
It's used here:
https://github.com/ZcashFoundation/zebra/pull/8923/files#diff-4f5cabe26761257a4d685a6edc7a43e0fe0f78762f50eeb48530f2bd3b3ee7caR81
and here:
https://github.com/ZcashFoundation/zebra/pull/8923/files#diff-4f5cabe26761257a4d685a6edc7a43e0fe0f78762f50eeb48530f2bd3b3ee7caR100

From[ our documentation](https://github.com/ZcashFoundation/zebra/blob/46c6b6eb38617c07e8a130650b33831759b73263/book/src/user/docker.md?plain=1#L59), if we do the following:
```shell
docker build -f ./docker/Dockerfile --target runtime --build-arg FEATURES='default-release-binaries prometheus' --tag local/zebra.mining:latest .
```

The `entrypoint` will evaluate the `$FEATURES` var and it will be empty, as it was never defined as a variable. This would be confusing for the user as they built it with `--build-arg FEATURES='default-release-binaries prometheus'` but the configuration file is not adding the corresponding section, as that argument was not passed as an Environment variable (at build time) to the container.